pallets/loans/src/lib.rs

```diff
@@ -914,12 +914,13 @@ impl<T: Config> Pallet<T> {
         if deposits.voucher_balance.is_zero() {
             return Ok(FixedU128::zero());
         }
+        let exchange_rate = Self::exchange_rate(asset_id);
+        let underlying_amount =
+            Self::calc_underlying_amount(deposits.voucher_balance, exchange_rate)?;
         let market = Self::market(asset_id)?;
-        let collateral_amount = Self::exchange_rate(asset_id)
-            .checked_mul_int(market.collateral_factor.mul_floor(deposits.voucher_balance))
-            .ok_or(ArithmeticError::Overflow)?;
+        let effects_amount = market.collateral_factor.mul_ceil(underlying_amount);
```

Comment:
so the point is `mul_ceil` right?

Reply:
we should maybe check `mul_floor`, `mul_ceil` and understand each use case

Reply:
Yes, otherwise the test will go fail. 
https://github.com/parallel-finance/parallel/pull/839/files#diff-47f90f2e22bcae2eb1e7e0b3876dafa029fc292bf8f605a2b600aa56062bc517R73